cishouseholds/validate.py

```diff
@@ -185,6 +187,65 @@ def check_singular_match(
     return df.drop("TOTAL")
 
 
+class ConfigError(Exception):
+    pass
+
+
+def upfront_key_value_parameters_validation(all_function_dict: Dict, config_file_arguments_list: List):
+    """
+    Checks that there's a valid input in the pipeline_config.yaml for every stage
+    input argument.
+
+    Parameters
+    ----------
+    all_function_dict: dictionary of all functions name and function object in pipeline_stages.py
+    pipeline_stage_list: from the config file all the functions that have been set up to run.
+    """
+    error_msg = ""
+    for stage_dict in config_file_arguments_list:
+        if type(stage_dict["run"]) != bool:
```

Comment:
I think you should first check if "run" exists, otherwise this will raise an exception when you try to select it from the dict

"run" is mandatory, so we should let them know if it's missing for any function

Reply:
similarly, I think we should validated that each one has a "function" specified, as this is mandatory